js/hotgraphicView.js

```diff
@@ -0,0 +1,216 @@
+
+            var model = this.prepareNarrativeModel();
+            var newNarrative = new NarrativeView({ model: model });
+            var $container = $(".component-container", $("." + this.model.get("_parentId")));
+
+            newNarrative.reRender();
+            newNarrative.setupNarrative();
+            $container.append(newNarrative.$el);
+            Adapt.trigger('device:resize');
+            _.defer(_.bind(function () {
+                this.remove();
+            }, this));
+        },
+
+        prepareNarrativeModel: function() {
+            var model = this.model;
+            model.set({
+                '_component': 'narrative',
+                '_wasHotgraphic': true,
+                'originalBody': model.get('body'),
+                'originalInstruction': model.get('instruction')
+            });
+
+            // Check if active item exists, default to 0
+            var activeItem = model.getActiveItem();
+            if (!activeItem) {
+                model.getItem(0).toggleActive(true);
+            }
+
+            // Swap mobile body and instructions for desktop variants.
+            if (model.get('mobileBody')) {
+                model.set('body', model.get('mobileBody'));
+            }
+            if (model.get('mobileInstruction')) {
+                model.set('instruction', model.get('mobileInstruction'));
+            }
+
+            return model;
+        },
+
+        onItemsActiveChange: function(model, _isActive) {
+            this.getItemElement(model).toggleClass('active', _isActive);
+        },
+
+        getItemElement: function(model) {
+            var index = model.get('_index');
+            return this.$('.hotgraphic-graphic-pin').filter('[data-index="' + index + '"]');
+        },
+
+        onItemsVisitedChange: function(model, _isVisited) {
+            if (!_isVisited) return;
+            var $pin = this.getItemElement(model);
+
+            // Append the word 'visited.' to the pin's aria-label
+            var visitedLabel = this.model.get('_globals')._accessibility._ariaLabels.visited + ".";
+            $pin.attr('aria-label', function(index, val) {
+                return val + " " + visitedLabel;
+            });
+
+            $pin.addClass('visited');
+        },
+
+        // Used to check if the hotgraphic should reset on revisit
+        checkIfResetOnRevisit: function() {
+            var isResetOnRevisit = this.model.get('_isResetOnRevisit');
+
+            // If reset is enabled set defaults
+            if (isResetOnRevisit) {
+                this.model.reset(isResetOnRevisit);
+            }
+        },
+
+        preRender: function() {
+            if (Adapt.device.screenSize === 'large') {
+                this.render();
+            } else {
+                this.reRender();
+            }
+        },
+
+        postRender: function() {
+            this.renderState();
+            this.$('.hotgraphic-widget').imageready(_.bind(function() {
```

Comment:
and this one can be changed to `this.$('.hotgraphic-widget').imageready(this.setReadyStatus.bind(this));`